src/commands/chat/moderation/unban.ts

```diff
@@ -0,0 +1,156 @@
+        if (!guild) return;
+        await guild.bans.fetch();
+
+        const modlogEmbeds: MessageEmbed[] = [];
+        const unbanReason = args.getString('reason') ?? 'No reason provided.';
+        const providedUserId = args.getString('user_id', true);
+        const bannedUser = guild.bans.cache.get(providedUserId);
+
+        if (!bannedUser) {
+            this.sendErrorEmbed(interaction, 'User Not Banned', `Unable to ban user ID ${providedUserId} - they are not banned.`);
+        }
+
+        const unbanUserInfo = await UserModel.findOne({ discordId: providedUserId });
+        const possibleAlts = unbanUserInfo && unbanUserInfo.uwid ? await UserModel.find({ uwid: unbanUserInfo.uwid }) : [];
+
+        if (unbanUserInfo && unbanUserInfo.uwid) {
+            await BanModel.updateMany(
+                { discordId: [providedUserId, ...possibleAlts.map((alt) => alt.uwid).filter((uwid) => !!uwid)] },
+                { unbanned: true }
+            );
+        }
+
+        for (const alt of possibleAlts) {
+            const altMember = guild.bans.cache.get(alt.discordId);
+            if (altMember) {
+                const altUnbanMessage = `Unbanned by ${this.getUser(interaction).tag} | Alt of ${
+                    bannedUser?.user.tag ?? `user with id ${providedUserId}`
+                } | ${unbanReason}`;
+
+                await guild.bans.remove(alt.discordId, altUnbanMessage);
+
+                modlogEmbeds.push(
+                    Modlog.getUserEmbed(
+                        altMember.user,
+                        `
+                            **User**: ${altMember}
+                            **Action**: Unban
+                            **Reason**: ${unbanReason} (alt of ${bannedUser?.user.tag ?? `user with id ${providedUserId}`})
+                            **Moderator**: ${interaction.member}
+                        `,
+                        'GREEN'
+                    )
+                );
+            }
+        }
+
+        if (bannedUser) {
+            modlogEmbeds.unshift(
+                Modlog.getUserEmbed(
+                    bannedUser.user,
+                    `
+                    **User**: ${bannedUser}
+                    **Action**: Ban
+                    **Reason**: ${unbanReason}
+                    **Moderator**: ${interaction.member}
+                    `,
+                    'GREEN'
+                )
+            );
+
+            await guild.bans.remove(providedUserId, `Unbanned by ${this.getUser(interaction).tag} | ${unbanReason}`);
+
+            await interaction.reply({
+                embeds: [
+                    new MessageEmbed()
+                        .setDescription(
+                            `**User ID ${providedUserId} and and ${modlogEmbeds.length - 1} alt accounts were unbanned |** ${unbanReason}`
+                        )
+                        .setColor('GREEN'),
+                ],
+            });
+        } else {
```

Comment:
This block will trigger if the user isn't banned but they had alt accounts that were banned which I don't believe is intended